File: backend/app/schemas/admin_collector.py
"""Secret-free administrator collector status response contracts."""

from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel


class CollectorQueueStatus(BaseModel):
    queue_name: str
    broker_available: bool
    worker_available: bool
    worker_count: int


class CollectorScheduleStatus(BaseModel):
    enabled: bool
    source_id: str
    requested_count: int
    complete_snapshot: bool
    cron_hour: int
    cron_minute: int
    timezone: str


class CollectorRunSummary(BaseModel):
    run_id: UUID
    status: str
    trigger_type: str
    started_at: datetime
    finished_at: datetime | None
    is_stale: bool
    requested_count: int
    inserted_count: int
    updated_count: int
    failed_count: int
    error_type: str | None


class AdminCollectorStatus(BaseModel):
    source_id: str
    display_name: str
    source_type: Literal["api", "file", "web"]
    manual_run_enabled: bool
    runtime_status: Literal[
        "ready",
        "configuration_required",
        "unavailable",
        "unknown",
    ]
    worker_registered: bool | None
    credential_status: Literal[
        "configured",
        "missing",
        "not_required",
        "unknown",
    ]
    public_policy_count: int
    active_run: CollectorRunSummary | None
    last_run: CollectorRunSummary | None


class AdminCollectorStatusResponse(BaseModel):
    generated_at: datetime
    queue: CollectorQueueStatus
    schedule: CollectorScheduleStatus
    collectors: list[AdminCollectorStatus]

File: frontend/src/types/adminCollector.ts
import type {
  CollectionRunStatus,
  CollectionRunTriggerType,
} from './collectionRun.js';

export const ADMIN_COLLECTORS_PATH = '/api/v1/admin/collectors';

export type CollectorSourceType = 'api' | 'file' | 'web';
export type CollectorRuntimeStatus =
  | 'ready'
  | 'configuration_required'
  | 'unavailable'
  | 'unknown';
export type CollectorCredentialStatus =
  | 'configured'
  | 'missing'
  | 'not_required'
  | 'unknown';

export interface CollectorQueueStatusDto {
  queue_name: string;
  broker_available: boolean;
  worker_available: boolean;
  worker_count: number;
}

export interface CollectorScheduleStatusDto {
  enabled: boolean;
  source_id: string;
  requested_count: number;
  complete_snapshot: boolean;
  cron_hour: number;
  cron_minute: number;
  timezone: string;
}

export interface CollectorRunSummaryDto {
  run_id: string;
  status: CollectionRunStatus;
  trigger_type: CollectionRunTriggerType;
  started_at: string;
  finished_at: string | null;
  is_stale: boolean;
  requested_count: number;
  inserted_count: number;
  updated_count: number;
  failed_count: number;
  error_type: string | null;
}

export interface AdminCollectorStatusDto {
  source_id: string;
  display_name: string;
  source_type: CollectorSourceType;
  manual_run_enabled: boolean;
  runtime_status: CollectorRuntimeStatus;
  worker_registered: boolean | null;
  credential_status: CollectorCredentialStatus;
  public_policy_count: number;
  active_run: CollectorRunSummaryDto | null;
  last_run: CollectorRunSummaryDto | null;
}

export interface AdminCollectorStatusResponse {
  generated_at: string;
  queue: CollectorQueueStatusDto;
  schedule: CollectorScheduleStatusDto;
  collectors: AdminCollectorStatusDto[];
}
